src/System.Linq/src/System/Linq/SelectMany.cs

```diff
@@ -231,18 +231,10 @@ public TResult[] ToArray()
                 {
                     IEnumerable<TResult> enumerable = _selector(element);
 
-                    int count;
-                    if (EnumerableHelpers.TryGetCount(enumerable, out count))
+                    if (builder.ReserveOrAdd(enumerable))
```

Comment:
I introduced this method to help with `Concat` (it's used there) then I realized it could be used in `SelectMany` too.